src/panels/lovelace/cards/hui-picture-glance-card.ts

```diff
@@ -0,0 +1,264 @@
+
+interface Config extends LovelaceConfig {
+  entities: EntityConfig[];
+  title?: string;
+  navigation_path?: string;
+  image?: string;
+  camera_image?: string;
+  state_image?: {};
+  aspect_ratio?: string;
+  entity?: string;
+  force_dialog?: boolean;
+}
+
+class HuiPictureGlanceCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+  private _entitiesDialog?: EntityConfig[];
+  private _entitiesToggle?: EntityConfig[];
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 3;
+  }
+
+  public setConfig(config: Config) {
+    if (
+      !config ||
+      !config.entities ||
+      !Array.isArray(config.entities) ||
+      !(config.image || config.camera_image || config.state_image) ||
+      (config.state_image && !config.entity)
+    ) {
+      throw new Error("Invalid card configuration");
+    }
+
+    const entities = processConfigEntities(config.entities);
+    this._entitiesDialog = [];
+    this._entitiesToggle = [];
+
+    entities.forEach((item) => {
+      if (
+        config.force_dialog ||
+        !DOMAINS_TOGGLE.has(computeDomain(item.entity))
+      ) {
+        this._entitiesDialog!.push(item);
+      } else {
+        this._entitiesToggle!.push(item);
+      }
+    });
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const isClickable =
+      this._config.navigation_path || this._config.camera_image;
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        <hui-image
+          class="${
+            classMap({
+              clickable: Boolean(isClickable),
+            })
+          }"
+          @click="${this._handleImageClick}"
+          .hass="${this.hass}"
+          .image="${this._config.image}"
+          .stateImage="${this._config.state_image}"
+          .cameraImage="${this._config.camera_image}"
+          .entity="${this._config.entity}"
+          .aspectRatio="${this._config.aspect_ratio}"
+        ></hui-image>
+        <div class="box">
+          ${
+            this._config.title
+              ? html`
+                  <div class="title">${this._config.title}</div>
+                `
+              : ""
+          }
+          <div>
+            ${
+              this._entitiesDialog!.filter(
+                (entityConf) => entityConf.entity in this.hass!.states
+              ).map((entityConf) => {
+                return html`
+                  <ha-icon
+                    .entity="${entityConf.entity}"
+                    @click="${this._openDialog}"
+                    class="${
+                      classMap({
+                        "state-on": !STATES_OFF.has(
+                          this.hass!.states[entityConf.entity].state
```

Comment:
This can be undefined.

Reply:
It might be easier to extract the render logic for each entity into a function so you can do checks if state obj exists etc.